syntax = "proto3";

package netbox;

// ===== Текущий функционал =====

message Target {
  string target = 1;       // Идентификатор цели (IP, hostname, префикс)
  string target_type = 2;  // Тип цели ("device", "subnet", "ip")
}

message GetTargetsRequest {}

message GetTargetsResponse {
  repeated Target targets = 1;
}

// ===== Новые сообщения =====

message Device {
  string name = 1;
  string platform = 2;
  string model = 3;
  string ip = 4;
}

message Subnet {
  string prefix = 1;
  string description = 2;
  string vlan = 3;
}
message IPAddress {
  string address = 1;
  string dns_name = 2;
  string device_name = 3;
  string interface_name = 4;
}

message GetDevicesRequest {
  string name = 1;
}

message GetDevicesResponse {
  repeated Device devices = 1;
}

message GetSubnetsRequest {
  string prefix = 1;
}

message GetSubnetsResponse {
  repeated Subnet subnets = 1;
}

message GetIPsRequest {
  string address = 1;
}

message GetIPsResponse {
  repeated IPAddress ips = 1;
}

// ===== gRPC сервис =====

service NetBox {
  rpc GetTargets(GetTargetsRequest) returns (GetTargetsResponse);

  rpc GetDevices(GetDevicesRequest) returns (GetDevicesResponse);
  rpc GetSubnets(GetSubnetsRequest) returns (GetSubnetsResponse);
  rpc GetIPs(GetIPsRequest) returns (GetIPsResponse);
}